feat: add documentation about authentication providers (#3)
ICIJ/datashare-docs: server-mode/about-the-server-mode.md

# About the server mode

In **server mode**, Datashare operates as a centralized server-based system. Users can access to the platform through a web interface, and the documents are stored and processed on Datashare's servers. This mode offers the advantage of easy accessibility from anywhere with an internet connection, as users can log in to the platform remotely. It also facilitate seamless collaboration among users, as all the documents and analysis are centralized.

# Server configuration

Datashare is launched with `--mode SERVER` and you have to provide:

- the external elasticsearch index address `elasticsearchAddress`
- a Redis store address `redisAddress`
- a Redis data bus address `messageBusAddress`
- a database [JDBC URL](https://docs.oracle.com/cd/E57185_01/ESTUG/apbs02s04s01.html) `dataSourceUrl`

> - the host of datashare (for batch search results URL generation) `rootHost`

Reviewer: ```suggestion
- the host of Datashare (used to generate batch search results URLs) `rootHost`
```
